docs: align README with the standard model (#19)
canonical/pydantic-kitbash: README.md

# pydantic-kitbash

> Kitbash is a Sphinx extension that automates the generation of reference documentation

Reviewer: Have we triple-checked that "Kitbash" isn't used anywhere else in the Sphinx world?

Author: There are no other packages named "kitbash" on PyPI, and I have yet to find a webpage including both "Sphinx" and "Kitbash". :+1: 

I can't _guarantee_ that the name is unique in this context, but I'm fairly confident.

for Pydantic models.

> As it traverses a model, Kitbash collects field data and generates consistent,
> well-formed output to be included in your product's documentation. To supplement this
> information, you can add Python docstrings below fields, which Kitbash will parse as
> reStructuredText to be included in the field's description.

Reviewer: ```suggestion
Kitbash parses a model to describe its fields in a Sphinx document. It can
target an entire model or specific fields. When covering a specific field, you can add
supplemental reStructuredText after the automatic portion of the text.
```

Author: https://github.com/canonical/pydantic-kitbash/pull/19/commits/9f5153938e64c704a949477ca7d4144473e14600

I changed the last sentence so that it's clear that the supplemental info goes in the docstring rather than the rST file.

Reviewer: Oh oops, I completely misread that sentence. Morning brain.

## Basic usage

> To document an individual field, add the `kitbash-field` directive to your rST document:

Reviewer: ```suggestion
To document an individual field, add the `kitbash-field` directive to your document:
```

Author: https://github.com/canonical/pydantic-kitbash/pull/19/commits/9f5153938e64c704a949477ca7d4144473e14600

```
.. kitbash-field:: <model-name> <field-name>
```

If you'd prefer to document an entire model, add the `kitbash-model` directive to your

> rST document:

Reviewer: ```suggestion
document:
```

Author: https://github.com/canonical/pydantic-kitbash/pull/19/commits/9f5153938e64c704a949477ca7d4144473e14600

```
.. kitbash-model:: <model-name>
```

> This will create an entry for each of the model's fields in the order they're defined.

Reviewer: I'd see if I could combine this with the previous paragraph.

Author: https://github.com/canonical/pydantic-kitbash/pull/19/commits/9f5153938e64c704a949477ca7d4144473e14600

I ended up cutting this out. I don't think it needs to be mentioned explicitly in the README.

## Project setup

To add Kitbash to your project, add `pydantic-kitbash` to the appropriate dependency

> group in your pyproject.toml file:

Reviewer: ```suggestion
group in your `pyproject.toml` file:
```

Author: https://github.com/canonical/pydantic-kitbash/pull/19/commits/9f5153938e64c704a949477ca7d4144473e14600

> group in your pyproject.toml file:

Reviewer: There are too many ways to set up projects in Python. I think I'd generalise this paragraph but state that `pyproject.toml` and the dependency group is an example case.

Author: If we're generalizing this, would it be better if I just state that it can be installed with pip?

Reviewer: That sounds best, yes. Pip is the default, for better or worse.

```toml
[dependency-groups]
docs = [
    "pydantic-kitbash",

>     ...

Reviewer 2: ```suggestion
    ...
    # ...
```

This leaves an ugly red highlight on the rendered document, commenting it out would be better (and I'd maybe comment out the Python one too just for consistency's sake)

Reviewer: We could probably cut the ellipses.

Author: https://github.com/canonical/pydantic-kitbash/pull/19/commits/9f5153938e64c704a949477ca7d4144473e14600

]
```

After adding Kitbash to your Python project, update Sphinx's `conf.py` file to include

> Kitbash:

Reviewer: ```suggestion
Kitbash as one of its extensions:
```

Author: https://github.com/canonical/pydantic-kitbash/pull/19/commits/9f5153938e64c704a949477ca7d4144473e14600

```python
extensions = [
    "pydantic_kitbash",

>     ...

Reviewer: ```suggestion
```

Author: https://github.com/canonical/pydantic-kitbash/pull/19/commits/9f5153938e64c704a949477ca7d4144473e14600

]
```

## Community and support

You can report any issues or bugs on the project's [GitHub
repository](https://github.com/canonical/pydantic-kitbash/issues).

Kitbash is covered by the [Ubuntu Code of
Conduct](https://ubuntu.com/community/ethos/code-of-conduct).

## License and copyright

Kitbash is released under the [LGPL-3.0 license](LICENSE).

> @2025 Canonical Ltd.

Reviewer: ```suggestion
© 2025 Canonical Ltd.
```
